pyquil/api/compiler.py

```diff
@@ -161,3 +166,69 @@ def get_job_fn():
         return wait_for_job(get_job_fn,
                             ping_time if ping_time else self.ping_time,
                             status_time if status_time else self.status_time)
+
+    def _pipo_payload(self, clifford, pauli):
+        """
+        Prepares a JSON payload for conjugating a Pauli by a Clifford - see apply_clifford_to_pauli.
+
+        :param Program clifford: A Program that consists only of Clifford operations.
+        :param PauliTerm pauli: A PauliTerm to be acted on by clifford via conjugation.
+        :return: The JSON payload, with keys "clifford" and "pauli".
+        """
+        max_qubit = max(max(pauli.get_qubits()), max(clifford.get_qubits()))
+        pauli_string = ["I" for _ in range(max_qubit + 1)]
+        for index, term in pauli.operations_as_set():
+            pauli_string[int(index)] = term
+        pauli_string = "".join(pauli_string)
+        payload = {"clifford": clifford.out(),
+                   "pauli": pauli_string}
+        return payload
+
+    def apply_clifford_to_pauli(self, clifford, pauli_in):
+        """
+        Given a circuit that consists only of elements of the Clifford group, return its action on a PauliTerm.
+
+        In particular, for Clifford C, and Pauli P, this returns the PauliTerm representing PCP^{\dagger}.
+
+        :param Program clifford: A Program that consists only of Clifford operations.
+        :param PauliTerm pauli_in: A PauliTerm to be acted on by clifford via conjugation.
+        :return: A PauliTerm corresponding to pauli_in * clifford * pauli_in^{\dagger}
+        """
+        payload = self._pipo_payload(clifford, pauli_in)
+        phase_factor, paulis = post_json(self.session, self.async_endpoint + "/pipo", payload).json()
+        pauli_out = PauliTerm("I", 0, 1.j ** phase_factor)
+        for i, pauli in enumerate(paulis):
+            pauli_out *= PauliTerm(pauli, i)
+        return pauli_out * pauli_in.coefficient
```

Comment:
This will always return a pauliterm operating on qubits 0...N independent of the qubits used in the program

Reply:
Oh geez. I think this has avoided being a problem because the representation I've been using just references everything to 0, but I'm going to fix that, so this will need to be changed too.

Reply:
I'm going to change the backend first before I change this.